src/main/java/uk/gov/justice/hmpps/prison/repository/OffenderCurfewRepository.java

```diff
@@ -186,20 +202,73 @@ public Optional<HomeDetentionCurfew> getLatestHomeDetentionCurfew(long bookingId
-        val results = jdbcTemplate.query(
+        return jdbcTemplate.query(
                 OffenderCurfewRepositorySql.LATEST_BATCH_HOME_DETENTION_CURFEW.getSql(),
                 createParams(
                   "bookingIds", bookingIds,
                   "statusTrackingCodes", statusTrackingCodesToMatch),
                   HOME_DETENTION_CURFEW_ROW_MAPPER
                 );
-        return results;
     }
 
 
     public void resetCurfew(long curfewId) {
+        getCurfewLock(curfewId);
         jdbcTemplate.update(
                 OffenderCurfewRepositorySql.RESET_OFFENDER_CURFEW.getSql(),
                 createParams("curfewId", curfewId));
     }
+
+    private void getCurfewLock(long curfewId) {
+        try {
+            jdbcTemplate.query(
+                OffenderCurfewRepositorySql.GET_OFFENDER_CURFEW_LOCK.getSql() + getWaitClauseIfNotHsqldb(),
+                createParams("curfewId", curfewId,
+                    "waitSeconds", lockWaitTime),
+                rs -> {
+                }
+            );
+        } catch (DataAccessException e) {
+            log.error("Error getting lock", e);
+            throw new DatabaseRowLockedException("Failed to get OFFENDER_CURFEWS lock for curfew id " + curfewId);
+        }
+    }
+
+    private void getTrackingsLock(long curfewId, Set<String> statusTrackingCodesToMatch) {
+        try {
+            jdbcTemplate.query(
+                OffenderCurfewRepositorySql.GET_HDC_STATUS_TRACKINGS_LOCK.getSql() + getWaitClauseIfNotHsqldb(),
+                createParams("curfewId", curfewId,
+                    "codes", statusTrackingCodesToMatch,
+                    "waitSeconds", lockWaitTime),
+                rs -> {
+                }
+            );
+        } catch (DataAccessException e) {
+            log.error("Error getting lock", e);
+            throw new DatabaseRowLockedException("Failed to get HDC_STATUS_TRACKINGS lock for curfew id " + curfewId);
+        }
+    }
+
+    private void getReasonsLock(long curfewId, Set<String> statusTrackingCodesToMatch) {
+        try {
+            jdbcTemplate.query(
+                OffenderCurfewRepositorySql.GET_HDC_STATUS_REASONS_LOCK.getSql() + getWaitClauseIfNotHsqldb(),
+                createParams("curfewId", curfewId,
+                    "codes", statusTrackingCodesToMatch,
+                    "waitSeconds", lockWaitTime),
+                rs -> {
+                }
+            );
+        } catch (DataAccessException e) {
+            log.error("Error getting lock", e);
+            throw new DatabaseRowLockedException("Failed to get HDC_STATUS_REASONS lock for curfew id " + curfewId);
+        }
+    }
+
+    @NotNull
+    private String getWaitClauseIfNotHsqldb() {
+        final HikariDataSource resolvedDefaultDataSource = (HikariDataSource)routingDataSource.getResolvedDefaultDataSource();
+        return resolvedDefaultDataSource.getJdbcUrl().contains("hsqldb") ? "" : " WAIT " + lockWaitTime;
```

Comment:
very minor - I think I would have done this nasty if hsqldb stuff using Spring beans, like the way we swap certain beans depending if we are using NOMIS or hsqldb e.g https://github.com/ministryofjustice/prison-api/blob/7aceba7c8c6b9ebd675f0fb259c4c56f0f2acd63/src/main/java/uk/gov/justice/hmpps/prison/service/receiveandtransfer/TrustAccountService.kt

Reply:
Ah ok yes Ill do it the same way as this